sycl/include/sycl/handler.hpp

```diff
@@ -752,10 +749,17 @@ class __SYCL_EXPORT handler {
     // header, so don't perform things that require it.
     if (KernelHasName) {
       // TODO support ESIMD in no-integration-header case too.
+
       clearArgs();
-      extractArgsAndReqsFromLambda(MHostKernel->getPtr(),
-                                   detail::getKernelParamDescs<KernelName>(),
-                                   detail::isKernelESIMD<KernelName>());
+      size_t NumParams = detail::getKernelNumParams<KernelName>();
+      reserveArgs(NumParams);
+
+      for (size_t I = 0, IndexShift = 0; I < NumParams; ++I) {
+        extractArgsAndReqsFromLambda(
```

Comment:
```suggestion
        extractArgsAndReqsFromLambda(
```

Reply:
Thank you for the advice, I moved the loop over args inside of `extractArgsAndReqsFromLambda()`.